### ISSUE TITLE
Configuration loader overrides nested values

### ISSUE THREAD

**Issue by tox2ik** [laravel/framework#13143]
Given a scenario where nested configuration files are used such that there are
several config files (`#1`), the method `Bootstrap / LoadConfiguration ::
getConfigurationFiles` does not return the files in any particular order
leading to a potential situation where the files array may contain keys as 
listed in `#2`. When the loadConfigurationFiles method iterates over these
keys, we end up with several invocations to Config/Repository::set (`#3`).
The issue is that the last invocation will override the "nested" values
from the previous two.

If this scenario is indeed supported, as I would guess from reading `Bootstrap
/ LoadConfiguration :: getConfigurationNesting`, then a possible solution would
be to `ksort` the $files array in `getConfigurationFiles` or-and merge the
values in loadConfigurationFiles.  Merging the values is not desireable because
the order of the files is haphazard unless the filesystem makes some guarantees
there.

So to summarize; call ksort($files) in getConfigurationFiles() before 
`return $files`.

This was found in Laravel 5.1.

## "nested" configuration files 

    (#1)
    config/zumba.php
    config/zumba/classes.php
    config/zumba/instructors.php

## $files array in getConfigurationFiles()

    (#2)
    [
        zumba.classes => /full/path/zumba/classes.php
        zumba.instructors => /full/path/zumba/classes.php
        zumba => /full/path/zumba.php
    ]

## problematic invocation of Config\Repository::set

    (#3)
    $repository->set('zumba.classes', require ...)
    $repository->set('zumba.instructors', require ...)
    $repository->set('zumba', require ...)

**Comment by GrahamCampbell** [comment]
This is intended.
